bin/remote-prover/src/utils.rs

```diff
@@ -1,52 +1,124 @@
 use std::net::TcpListener;
 
+use http::HeaderMap;
 use miden_remote_prover::error::RemoteProverError;
 use pingora::{Error, ErrorType, http::ResponseHeader, protocols::http::ServerSession};
 use pingora_proxy::Session;
+use prost::Message;
+use tonic::Code;
 use tracing::debug;
 
 use crate::{COMPONENT, commands::PROXY_HOST, proxy::metrics::QUEUE_DROP_COUNT};
 
-const RESOURCE_EXHAUSTED_CODE: u16 = 8;
+/// Write a protobuf message as a gRPC response to a Pingora session
+///
+/// This helper function takes a protobuf message and writes it to a Pingora session
+/// in the proper gRPC format, handling message encoding, headers, and trailers.
+pub async fn write_grpc_response_to_session<T>(
+    session: &mut Session,
+    message: T,
+    grpc_status: Option<&str>,
```

Comment:
Why do we pass the `grpc_status` here? Doesn't seem like we actually ever use it.

Also, base don how we use it below in the function, if we do want to keep it as a parameter, should we not pass it as `Code`?